### ISSUE TITLE
hard-coded rush-repos directory

### ISSUE THREAD

**Issue by pcrockett** [DannyBen/rush-cli#77]
I was looking through how this is implemented before using it, and I have to say, it's really nice.

One little nit from me though: hard-coded `$HOME/rush-repos` path. It would be nice if this were configurable so I can put it with my other repos, or perhaps in `$XDG_DATA_HOME` (usually `$HOME/.local/share`).

Any chance this could be a setting in the config file?

**Comment by DannyBen** [comment]
I am happy you like it. I am using it heavily for [everything](https://github.com/DannyBen/rush-repo), everywhere.

If you want to review just the essence of the code without the surrounding CLI fluff, see the [src directory](https://github.com/DannyBen/rush-cli/tree/master/src).

The default location for repos (`$HOME/rush-repos`) has very little meaning, since it is the `$HOME/rush.ini` config that specifies which repo is located in which folder. Also note that you can have different rush repos scattered across your system, and the config can point to all of them. So there is no "root" folder for repos, only a default location if one was not provided during the `rush clone` operation.

So the answer to this:

> Any chance this could be a setting in the config file?

is that it already is.

Isn't this sufficient?

As a side note, I usually avoid using any `XDG_*` variable, as it is not appropriate (and not present) in headless linux.

**Comment by pcrockett** [comment]
Sorry, not at computer now...

I want to say I found a couple places where this is relevant, but the one I remember is the [snatch command](https://github.com/DannyBen/rush-cli/blob/master/src/snatch_command.sh). I may be missing something, but I believe it'll leave the `rush-repos` directory sitting in my HOME no matter what I do.

Though on second thought, it might make sense to use `mktemp` for that command anyway, since the repo will be removed at the end.

**Comment by DannyBen** [comment]
> I believe it'll leave the `rush-repos` directory sitting in my HOME no matter what I do.

If it does, it is a bug.
The code explicitly removes the repo when done.

https://github.com/DannyBen/rush-cli/blob/e8d598f28bb798a3ab02711f0545aa360b0b1ff8/src/snatch_command.sh#L9-L15

**Comment by pcrockett** [comment]
Ok, sorry, back at computer now. Here's how to reproduce the bug (tested on my machine):

```bash
cd ~
rm -rf rush-repos  # just in case it's already there
rush snatch DannyBen/rush-cli sample-repo/hello
test ! -d rush-repos || echo "rush-repos dir is still there!"
```

The cloned repository is removed, yes, but the parent `rush-repos` dir is not.

**Comment by DannyBen** [comment]
> The cloned repository is removed, yes, but the parent rush-repos dir is not.

This is functioning as it is designed. Why is this an issue?

**Comment by pcrockett** [comment]
Because it's additional home directory clutter. Even tools that don't adhere to the XDG standard at least hide their random directories by prepending them with `.`, which is better than nothing.

But generally that's one of the big reasons the XDG standard exists: It makes it so the various tools on your machine put their assets in well-organized, non-conflicting locations that the user can optionally control.

Side note: One can easily support the XDG standard without requiring those variables to be defined, so it doesn't matter if it's running on a server or desktop machine. Ex:

data_dir="${XDG_DATA_HOME:-$HOME/.local/share/rush}"

In any case I'm not even asking for that though. A config option (or env variable) would be fine.

Though I understand this is a personal project; If you like your tool the way it is and this request would make you like it less, I respect that. I'm still thankful you've shared it with the world.

**Comment by DannyBen** [comment]
Convinced.

How about this:

1. The default directory will be something like `$RUSH_ROOT` or `$HOME/rush-repos`.
2. The `rush snatch` command will work in a tmp directory.

Works?

**Comment by pcrockett** [comment]
Sounds great. This, along with RUSH_CONFIG, allows me to put the config file in ~/.config/rush/rush.ini and other data in ~/.local/share/rush which makes rush behave consistently with other tools. Nice and neat.

And yes, the tmp directory also sounds like a good improvement.

**Comment by DannyBen** [comment]
Done in #78 in case you want to take a look (ignore all the updates to the man pages).

- [clone_command](https://github.com/DannyBen/rush-cli/pull/78/files#diff-d90a1a2b46fb1e78a86add584a328ccb19cd1ed689602ebf449005162bd26198)
- [initialize](https://github.com/DannyBen/rush-cli/pull/78/files#diff-1cf9e1a253e61753a7e40bcd7f28dc8cf7bd91dc45f3cd848aed1a08437c65db)
- [snatch_command](https://github.com/DannyBen/rush-cli/pull/78/files#diff-aa338f0647bd67636fd31e16c60f447fd96f65e2168c4b931fe3ca958bcc6368)

I will fix and update the tests before proceeding.

**Comment by DannyBen** [comment]
It is ready to be merged, pending your review.

**Comment by pcrockett** [comment]
I'm not a reviewer on that PR, so I'll just leave a comment here: looks great. thanks a lot.

**Comment by DannyBen** [comment]
Thanks. Released v0.7.13.

**Comment by pcrockett** [comment]
Tested! Works!
